staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -280,6 +280,49 @@ type credentials struct {
 	cert  *tls.Certificate `datapolicy:"secret-key"`
 }
 
+func (a *Authenticator) UpdateTransport(transport http.RoundTripper) (http.RoundTripper, error) {
+	tlsConfig, err := utilnet.TLSClientConfig(transport)
+	if err != nil {
+		return transport, err
+	}
+	tlsConfig.GetClientCertificate = func(*tls.CertificateRequestInfo) (*tls.Certificate, error) {
```

Comment:
Mutating the TLS config is not allowed because it is shared across multiple instances - i.e. it could already be in use for handling connections.

Reply:
this is copied verbatim from the existing code, the only difference is that current process is

Config -> modify config with exec -> generate transport

and this is

config -> generate transport -> exec modify the generated transport

The transport mutation is the same, just happens in a different stage of the process ... am I missing something?

Reply:
> this is copied verbatim from the existing code, the only difference is that current process is
> 
> Config -> modify config with exec -> generate transport
> 
> and this is
> 
> config -> generate transport -> exec modify the generated transport
> 
> The transport mutation is the same, just happens in a different stage of the process ... am I missing something?

Today it is:

`Config -> modify+cache -> transport` on first call, later calls it is `Config -> get transport from cache`

This change makes it:

`Config -> transport+cache -> modify` on first call, later calls its `Config -> get transport from cache -> modify`

Create a test with multiple go routines that build clients at different times with the race detector enabled to see the data races.

Reply:
ok, I see my mistake, thanks for taking the time to correct me